runner/operator/copy_outputs_operator/v2_1_0/copy_outputs_operator.py

```diff
@@ -0,0 +1,130 @@
+            run_ids
+        )
+        mapping_file = self.write_to_file("sample_mapping.txt", mapping_file_content)
+        pairing_file = self.write_to_file("sample_pairing.txt", pairing_file_content)
+        data_clinical_file = self.write_to_file("sample_data_clinical.txt", data_clinical_content)
+
+        input_json["meta"] = [mapping_file, pairing_file, data_clinical_file]
+
+        number_of_runs = len(run_ids)
+        name = "ARGOS COPY OUTPUTS %s runs [%s,..] " % (number_of_runs, run_ids[0])
+
+        app = self.get_pipeline_id()
+        pipeline = Pipeline.objects.get(id=app)
+        project_prefix = input_json["project_prefix"]
+        output_directory_prefix = get_output_directory_prefix(self.run_ids)
+
+        tags = {"run_ids": run_ids}
+
+        copy_outputs_job_data = {"app": app, "inputs": input_json, "name": name, "tags": tags}
+
+        """
+        If project_prefix and job_group_id, write output to a directory
+        that uses both
+
+        Also use argos version number for output instead of pipeline version
+        that's listed in Beagle
+        """
+
+        argos_run = Run.objects.get(id=run_ids[0])
+        argos_pipeline = argos_run.app
+
+        output_directory = None
+        if project_prefix:
+            tags["project_prefix"] = project_prefix
+            if self.job_group_id:
+                output_prefix = output_directory_prefix if output_directory_prefix else project_prefix
+                jg = JobGroup.objects.get(id=self.job_group_id)
+                jg_created_date = jg.created_date.strftime("%Y%m%d_%H_%M_%f")
+                output_directory = os.path.join(
+                    pipeline.output_directory, "argos", output_prefix, argos_pipeline.version, jg_created_date
+                )
+            copy_outputs_job_data["output_directory"] = output_directory
+        copy_outputs_job = [RunCreator(**copy_outputs_job_data)]
+        return copy_outputs_job
+
+    def write_to_file(self, fname, s):
+        """
+        Writes file to temporary location, then registers it to the temp file group
+        """
+        tmpdir = os.path.join(settings.BEAGLE_SHARED_TMPDIR, str(uuid.uuid4()))
+        Path(tmpdir).mkdir(parents=True, exist_ok=True)
+        output = os.path.join(tmpdir, fname)
+        with open(output, "w+") as fh:
+            fh.write(s)
+        os.chmod(output, 0o777)
+        self.register_tmp_file(output)
+        return {"class": "File", "location": "juno://" + output}
+
+    def register_tmp_file(self, path):
+        fname = os.path.basename(path)
+        temp_file_group = FileGroup.objects.get(slug="temp")
+        file_type = FileType.objects.get(name="txt")
+        try:
+            File.objects.get(path=path)
+        except:
+            print("Registering temp file %s" % path)
+            f = File(file_name=fname, path=path, file_type=file_type, file_group=temp_file_group)
+            f.save()
```

Comment:
This is a wrong way of creating the file. Without the FileMetadata these files will be unsearchable by FileRepostory. We should probably do something like this https://github.com/mskcc/beagle/blob/master/beagle_etl/jobs/metadb_jobs.py#L1101-L1104

Reply:
what metadata should go into that file, just project_prefix?

Reply:
okay added the changes in!